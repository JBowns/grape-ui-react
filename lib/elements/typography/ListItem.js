'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ListItem = undefined;

var _templateObject = _taggedTemplateLiteral(['\n  ', '\n  margin-bottom: ', ';\n'], ['\n  ', '\n  margin-bottom: ', ';\n']);

var _styledComponents = require('styled-components');

var _styledComponents2 = _interopRequireDefault(_styledComponents);

var _textStyles = require('./textStyles');

var _globalStyles = require('../../global-styles');

var _utils = require('./utils');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _taggedTemplateLiteral(strings, raw) { return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

var _getGlobalStyles = (0, _globalStyles.getGlobalStyles)(),
    gridSchema = _getGlobalStyles.grid;

var ListItem = exports.ListItem = _styledComponents2.default.li(_templateObject, _textStyles.textStylesBase, (0, _utils.getScaledSize)(gridSchema.gutter, 0.25));