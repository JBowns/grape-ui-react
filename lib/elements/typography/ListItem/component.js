'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ListItemComponent = undefined;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _styledHelpers = require('../../../utils/styledHelpers');

var _componentHelpers = require('../../../utils/componentHelpers');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

var propsToTrim = Object.assign({}, _styledHelpers.typography.propTypes);
var ListItemComponent = function ListItemComponent(_ref) {
  var children = _ref.children,
      props = _objectWithoutProperties(_ref, ['children']);

  return _react2.default.createElement(
    'li',
    (0, _componentHelpers.removeSomeProps)(props, Object.keys(propsToTrim)),
    children
  );
};
exports.ListItemComponent = ListItemComponent;
ListItemComponent.propTypes = {
  children: _propTypes2.default.any.isRequired
};