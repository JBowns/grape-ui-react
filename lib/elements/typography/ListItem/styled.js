'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ListItem = undefined;

var _templateObject = _taggedTemplateLiteral(['\n  ', '\n  ', '\n  ', '\n  ', '\n  ', '\n  ', '\n  ', '\n  ', '\n  ', '\n  margin-bottom: ', ';\n  '], ['\n  ', '\n  ', '\n  ', '\n  ', '\n  ', '\n  ', '\n  ', '\n  ', '\n  ', '\n  margin-bottom: ', ';\n  ']);

var _styledComponents = require('styled-components');

var _styledComponents2 = _interopRequireDefault(_styledComponents);

var _globalStyles = require('../../../global-styles');

var _utils = require('../utils');

var _textStyles = require('../textStyles');

var _component = require('./component');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _taggedTemplateLiteral(strings, raw) { return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

var _getGlobalStyles = (0, _globalStyles.getGlobalStyles)(),
    gutter = _getGlobalStyles.grid.gutter;

var ListItem = exports.ListItem = (0, _styledComponents2.default)(_component.ListItemComponent)(_templateObject, _textStyles.getFontFamily, _textStyles.getFontSize, _textStyles.getFontWeight, _textStyles.getLetterSpacing, _textStyles.getLineHeight, _textStyles.getFontStyle, _textStyles.getColor, _textStyles.getTextAlign, _textStyles.getTextDecoration, (0, _utils.getScaledSize)(gutter, 0.25));

ListItem.propTypes = Object.assign({}, _textStyles.typography.propTypes);