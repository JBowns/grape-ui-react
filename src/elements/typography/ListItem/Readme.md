`<ListItem>` is a direct child for `<List>`.  Please read the docs for `<List>` on how to use this.