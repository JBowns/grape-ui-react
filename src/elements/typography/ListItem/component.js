import React from 'react';
import PropTypes from 'prop-types';
import { spaceProps, typography } from 'src/utils/styledHelpers';
import { removeSomeProps } from 'src/utils/componentHelpers';

const propsToTrim = {
  ...spaceProps.propTypes,
  ...typography.propTypes,
};
export const ListItemComponent = ({ children, ...props }) => (
  <li {...removeSomeProps(props, Object.keys(propsToTrim))}>
    {children}
  </li>
);
ListItemComponent.propTypes = {
  children: PropTypes.any.isRequired,
};
