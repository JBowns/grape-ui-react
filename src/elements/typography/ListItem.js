import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { getGlobalStyles } from 'src/global-styles';
import { getScaledSize } from './utils';
import {
  getFontFamily,
  getFontSize,
  getFontWeight,
  getLetterSpacing,
  getLineHeight,
  getFontStyle,
  getColor,
  getTextAlign,
  getTextDecoration,
  typography,
} from './textStyles';
import { passThrough } from 'src/utils/componentHelpers';

const { grid: { gutter } } = getGlobalStyles();

const ListItemComponent = ({ children, ...props }) => (
  <li {...passThrough(ListItemComponent, props)}>
    {children}
  </li>
);
ListItemComponent.propTypes = {
  children: PropTypes.any.isRequired,
  ...typography.propTypes,
};

export const ListItem = styled(ListItemComponent)`
  ${getFontFamily}
  ${getFontSize}
  ${getFontWeight}
  ${getLetterSpacing}
  ${getLineHeight}
  ${getFontStyle}
  ${getColor}
  ${getTextAlign}
  ${getTextDecoration}
  margin-bottom: ${getScaledSize(gutter, 0.25)};
  `;

ListItem.propTypes = {
  ...typography.propTypes,
};
