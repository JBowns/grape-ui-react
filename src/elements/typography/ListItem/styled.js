import styled from 'styled-components';
import { getGlobalStyles } from 'src/global-styles';
import { fontWeight, space } from 'styled-system';
import {
  colorCore,
  defaultStylesBase,
  fontFamilyCore,
  fontSizeCore,
  fontStyleCore,
  letterSpacingCore,
  lineHeightCore,
  scaleFont,
  textAlignCore,
  textDecorationCore,
  typography,
} from 'src/utils/styledHelpers';
import { ListItemComponent } from './component';

const { grid: { gutter } } = getGlobalStyles();

export const ListItem = styled(ListItemComponent)`
  ${colorCore}
  ${fontFamilyCore}
  ${fontSizeCore}
  ${fontWeight}
  ${letterSpacingCore}
  ${lineHeightCore}
  ${fontStyleCore}
  ${space}
  ${textAlignCore}
  ${textDecorationCore}
`;

ListItem.propTypes = {
  ...typography.propTypes,
};

ListItem.defaultProps = {
  ...defaultStylesBase,
  mb: scaleFont(gutter, 0.25),
};
