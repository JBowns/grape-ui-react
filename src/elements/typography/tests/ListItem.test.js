import React from 'react';
import { ListItem } from '../ListItem';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

describe('ListItem Component base', () => {
  it('should return a ListItem object with base styling', () => {
    const component = renderer.create(<ListItem>Lorem Ipsum</ListItem>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
describe('ListItem Component specified styles', () => {
  it('should return a ListItem object with font-style: italic', () => {
    const component = renderer.create(<ListItem italic>Lorem Ipsum</ListItem>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should return a ListItem object with font-size: lg', () => {
    const component = renderer.create(<ListItem lg>Lorem Ipsum</ListItem>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should return a ListItem object with font-size: sm', () => {
    const component = renderer.create(<ListItem sm>Lorem Ipsum</ListItem>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should return a ListItem object with a base color (implicit)', () => {
    const component = renderer.create(<ListItem color="green">Lorem Ipsum</ListItem>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should return a ListItem object with a base color', () => {
    const component = renderer.create(<ListItem color="green.base">Lorem Ipsum</ListItem>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should return a ListItem object with a light color', () => {
    const component = renderer.create(<ListItem color="green.light">Lorem Ipsum</ListItem>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should return a ListItem object with a dark color', () => {
    const component = renderer.create(<ListItem color="green.dark">Lorem Ipsum</ListItem>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should return a ListItem object with a specified font-family', () => {
    const component = renderer.create(<ListItem fontFamily="monospace">Lorem Ipsum</ListItem>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should return a ListItem object with a specified font-weight', () => {
    const component = renderer.create(<ListItem fontWeight="bold">Lorem Ipsum</ListItem>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should return a ListItem object with a specified kerning', () => {
    const component = renderer.create(<ListItem kerning="2px">Lorem Ipsum</ListItem>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should return a ListItem object with a specified text-align', () => {
    const component = renderer.create(<ListItem textAlign="right">Lorem Ipsum</ListItem>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should return a ListItem object with a specified text-decoration', () => {
    const component = renderer.create(<ListItem textDecoration="underline">Lorem Ipsum</ListItem>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
