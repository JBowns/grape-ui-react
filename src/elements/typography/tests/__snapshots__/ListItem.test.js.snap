// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem Component should return a ListItem object with font-size: lg 1`] = `
.c0 {
  font-size: calc(1rem * 1.2);
  line-height: 1.5;
  text-align: inherit;
  margin-bottom: calc(1rem / 4);
}

<li
  className="c0"
>
  Lorem Ipsum
</li>
`;

exports[`ListItem Component should return a ListItem object with font-size: sm 1`] = `
.c0 {
  font-size: calc(1rem * .8);
  line-height: 1.5;
  text-align: inherit;
  margin-bottom: calc(1rem / 4);
}

<li
  className="c0"
>
  Lorem Ipsum
</li>
`;

exports[`ListItem Component should return a ListItem object with font-style: base 1`] = `
.c0 {
  font-size: calc(1rem * 1);
  line-height: 1.5;
  text-align: inherit;
  margin-bottom: calc(1rem / 4);
}

<li
  className="c0"
>
  Lorem Ipsum
</li>
`;

exports[`ListItem Component should return a ListItem object with font-style: italic 1`] = `
.c0 {
  font-size: calc(1rem * 1);
  line-height: 1.5;
  font-style: italic;
  text-align: inherit;
  margin-bottom: calc(1rem / 4);
}

<li
  className="c0"
>
  Lorem Ipsum
</li>
`;
